src/browserbehavior/DefaultBrowserBehavior.ts

```diff
@@ -64,7 +64,10 @@ export default class DefaultBrowserBehavior implements BrowserBehavior, Extended
     return this.browser.version;
   }
 
-  majorVersion(): number {
+  majorVersion(): number | null {
```

Comment:
Not sure if it's fine to extend this return type and only in default implementation. The interface is still having `number` as return type.
https://github.com/aws/amazon-chime-sdk-js/blob/0aabd5aa821051e02569a0fb77736224416cdf3b/src/browserbehavior/BrowserBehavior.ts#L4-L13

Reply:
I think you beat me to this by a few minutes :), I really appreciate you looking into this!

I tried out overriding the return type in a typescript playground

```
interface BrowserBehavior {
    majorVersion(): number
}

class DefaultBrowserBehavior implements BrowserBehavior {
    majorVersion(): number | null {
        return 1
    }
}
```

Typescript complains about it, but it might still work once it's transpiled to javascript.

```
Property 'majorVersion' in type 'DefaultBrowserBehavior' is not assignable to the same property in base type 'BrowserBehavior'.
  Type '() => number | null' is not assignable to type '() => number'.
    Type 'number | null' is not assignable to type 'number'.
      Type 'null' is not assignable to type 'number'.(2416)
```

[Could also consider a default value of like -1 for major version](https://github.com/aws/amazon-chime-sdk-js/pull/2686)?

Reply:
@michaelbachmann I think -1 makes more sense. Besides this we think it's better to add fall back value for `browser.version` and `browser.os` if their values is `null` from `detect()`.